dhcpv6/nclient6/client.go

```diff
@@ -250,7 +259,7 @@ func (c *Client) RapidSolicit(ctx context.Context, modifiers ...dhcpv6.Modifier)
 	if err != nil {
 		return nil, err
 	}
-	msg, err := c.SendAndRead(ctx, c.serverAddr, solicit, IsMessageType(dhcpv6.MessageTypeReply))
+	msg, err := c.SendAndRead(ctx, c.serverAddr, solicit, IsMessageType(dhcpv6.MessageTypeReply, dhcpv6.MessageTypeAdvertise))
```

Comment:
I don't think this is valid.

If the response is a Reply, we can just return like we already do.

But if the response is an Advertise, then we must send a Request next: https://github.com/insomniacslk/dhcp/blob/834a63bade8aa8927af8b4c9842f0ad0345289d4/dhcpv6/nclient6/client.go#L275

Reply:
The RFC is not specific about the behaviour in case of Advertise instead of Offer/Ack. If a server ignores the rapidcommit flag and treats it as a 4-way transaction, it's unspecified what the client should do. In this case I would just ignore the reply here, and let the logic around `RapidSolicit` to choose whether to retry with a 4-way transaction. What are your thoughts?

https://tools.ietf.org/html/rfc4039#section-3.1

Reply:
My thought is that's valid, but takes twice as long as you wait for RapidSolicit to time out.

Reply:
another option might be to let the function fall back to "relaxed" mode if the server doesn't respond. The reason I call it relaxed is that the answer is not valid (the server should not respond with advertise to a rapid-solicit, and should drop it if not able to handle that), but we can break the rules and make it work without waiting for timeout. But I'd like that the default stays "don't break the rule" and provide a flag to do otherwise. Thoughts?